src/core/search/search.cc

```diff
@@ -0,0 +1,25 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "core/search/search.h"
+
+#include "core/search/query_driver.h"
+
+using namespace std;
+
+namespace dfly::search {
+
+AstExpr ParseQuery(std::string_view query) {
+  QueryDriver driver{};
+  driver.ResetScanner();
+  driver.SetInput(std::string{query});
+  try {
+    (void)Parser (&driver)();
+  } catch (...) {
+    return {};
+  }
```

Comment:
Yes yes, I know its just duct tape to avoid crashing 🙂 Need to check how to get parser error back more in detail, out of scope for this part